dev-db/postgresql/postgresql-10.1.recipe

```diff
@@ -0,0 +1,147 @@
+	cmd:pg_resetwal$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:pg_resetxlog$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:pg_restore$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:pg_rewind$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:pg_test_fsync$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:pg_test_timing$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:pg_upgrade$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:pg_waldump$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:pgbench$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:postgres$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:postmaster$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:psql$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:reindexdb$secondaryArchSuffix = $portVersion compat >= 10
+	cmd:vacuumdb$secondaryArchSuffix = $portVersion compat >= 10
+	lib:libecpg$secondaryArchSuffix = $portVersion compat >= 10
+	lib:libecpg_compat$secondaryArchSuffix = $portVersion compat >= 10
+	lib:libpgtypes$secondaryArchSuffix = $portVersion compat >= 10
+	lib:libpq$secondaryArchSuffix = $portVersion compat >= 10
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libicuuc$secondaryArchSuffix
+	lib:libicui18n$secondaryArchSuffix
+	lib:libicudata$secondaryArchSuffix
+	lib:libreadline$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	lib:libxslt$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	postgresql${secondaryArchSuffix}_devel = $portVersion
+	devel:libpgport$secondaryArchSuffix = $portVersion
+	devel:libpgcommon$secondaryArchSuffix = $portVersion
+	devel:libecpg$secondaryArchSuffix = $portVersion
+	devel:libpgfeutils$secondaryArchSuffix = $portVersion
+	devel:libpgtypes$secondaryArchSuffix = $portVersion
+	devel:libpq$secondaryArchSuffix = $portVersion
+	devel:libecpg_compat$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	postgresql${secondaryArchSuffix} == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	devel:libicuuc$secondaryArchSuffix
+	devel:libicui18n$secondaryArchSuffix
+	devel:libicudata$secondaryArchSuffix
+	devel:libreadline$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	devel:libxslt$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:aclocal
+	cmd:awk
+	cmd:bison
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	CFLAGS=-D_BSD_SOURCE runConfigure ./configure \
+		--with-icu \
+		--with-libxml \
+		--with-libxslt
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	prepareInstalledDevelLibs libpgport \
+		libpgcommon \
+		libecpg \
+		libpgtypes \
+		libpq \
+		libecpg_compat \
+		libpgfeutils
+	fixPkgconfig
+
+	mkdir -p $developLibDir/postgresql/pgxs
+	mv $libDir/postgresql/pgxs $developLibDir/postgresql/
+
+	packageEntries devel \
+		$developDir
+}
+
+TEST()
+{
+	make check
```

Comment:
I haven't checked yet, but if the shared libs are built in e.g. `dev-db/postgresql/work-9.3.6/sources/postgresql-10.1/src/lib/` then this should work:
```diff
-	make check
+	LIBRARY_PATH="$sourceDir/src/lib${LIBRARY_PATH:+:$LIBRARY_PATH}" make check
```